lib/src/kvs/ds.rs

```diff
@@ -1292,3 +1383,78 @@ impl Datastore {
 		Ok(())
 	}
 }
+
+async fn catchup_live_queries(
+	mut tx: Transaction,
+	local_live_queries: Arc<RwLock<BTreeMap<LqIndexKey, LqIndexValue>>>,
+	notification_channel_sender: Sender<Notification>,
+	lqs_to_update: Vec<(LqIndexKey, LqIndexValue)>,
+	vs: Versionstamp,
+) {
+	for (k, v) in lqs_to_update {
+		// Process notifications
+		let changes = {
+			let changes = cf::read(
+				&mut tx,
+				&k.ns,
+				&k.db,
+				Some(&k.tb),
+				ShowSince::Versionstamp(conv::versionstamp_to_u64(&v.vs)),
+				Some(1000),
+			)
+			.await;
+			tx.cancel().await.expect("Error cancelling transaction for lq catchup");
+			changes.expect("Error reading change feed")
+		};
+
+		for database_change in changes {
+			for table_changes in database_change.1 .0 {
+				for table_change in table_changes.1 {
+					// TODO enforce security
```

Comment:
I feel like this is a big todo. Do we already have an issue to track this?

Reply:
Created SUR-291 and updated TODO comment